usr/share/rear/verify/DP/default/400_verify_dp.sh

```diff
@@ -1,17 +1,14 @@
 # 400_verify_dp.sh
-# read DP vars from config file
+# read Data Protector vars from config file
+
 CELL_SERVER="`cat /etc/opt/omni/client/cell_server`"
+OMNICHECK=/opt/omni/bin/omnicheck
 
-# check that cell server is actually available (ping)
+# the Cell Manager must be configured on the client
 [ "${CELL_SERVER}" ]
-StopIfError "Data Protector Cell Manager not set in /etc/opt/omni/client/cell_server (TCPSERVERADDRESS) !"
-
-if test "$PING" ; then
-	ping -c 1 "${CELL_SERVER}" >/dev/null 2>&1
-	StopIfError "Sorry, but cannot reach Data Protector Cell Manager ${CELL_SERVER}"
-
-	Log "Data Protector Cell Manager ${CELL_SERVER} seems to be up and running."
-else
-	Log "Skipping ping test"
-fi
+StopIfError "Data Protector Cell Manager TCPSERVERADDRESS not set in /etc/opt/omni/client/cell_server"
 
+# check that the Cell Manager is responding on the INET port
+${OMNICHECK} -patches -host ${CELL_SERVER}
+StopIfError "Data Protector Cell Manager is not responding, error code $?.
```

Comment:
A nice example why one should in general not use the old `StopIfError` function
because that function does not behave as one expects on first glance.

The `StopIfError` function is implemented in lib/_input-output-functions.sh as
```
# If return code is non-zero, bail out:
function StopIfError () {
    if (( $? != 0 )) ; then
        Error "$@"
    fi
}
```
https://github.com/rear/rear/blob/master/usr/share/rear/lib/_input-output-functions.sh#L691

Because of that indirection
cf. RFC 1925 item 6a https://tools.ietf.org/html/rfc1925
```
It is always possible to add another level of indirection.
```
using `$?` in the  `StopIfError` function message evaluates to a wrong value:
```
# cat QQQ ; if (( $? != 0 )) ; then echo "ERROR $?" ; fi

cat: QQQ: No such file or directory
ERROR 0
```

In contrast using bash directly
```
# cat QQQ || echo "ERROR $?"

cat: QQQ: No such file or directory
ERROR 1
```
works as is should.

This is one reason why I personally never use the `StopIfError` function.
Currently the `StopIfError` function is called 250 times in 124 scripts
and I don't find time to clean up all that places so the `StopIfError` function
is kept in ReaR.

In this particular case here I suggest to use plain bash e.g. like
```
${OMNICHECK} -patches -host ${CELL_SERVER} || Error "Data Protector Cell Manager is not responding (error code $?). See $RUNTIME_LOGFILE for details."
```
or to avoid too long lines (personally I don't mind long lines) something like
```
if ! ${OMNICHECK} -patches -host ${CELL_SERVER} ; then
    Error "Data Protector Cell Manager is not responding (error code $?). See $RUNTIME_LOGFILE for details."
fi
```